frontend/src/metabase/visualizations/components/settings/ChartSettingOrderedSimple.tsx

```diff
@@ -86,7 +84,7 @@ export const ChartSettingOrderedSimple = ({
         <ChartSettingOrderedItems
           items={orderedItems.map(item => ({
             ...item,
-            color: items[item.originalIndex].color,
+            color: items.find(i => i.key === item.key)?.color,
```

Comment:
nit: I think the `?` here is a bit misleading. Correct me if I'm wrong, but `items` and `orderedItems` should always contain the same amount of items and have the exact same set of keys. Otherwise, let say it's possible that we can't find an item from `items`, if we pass `color` as `undefined` there wouldn't be the color picker which is obviously wrong.

In the screenshot below, I purposely pass `2020`'s color as `undefined`
![image](https://user-images.githubusercontent.com/1937582/203261070-ecfcdcfe-6229-4ee7-90b1-c2bb9fb58a37.png)

So, I think it should be impossible that color is undefined here.

We should think of another way to satisfy TypeScript instead.

Reply:
It is really awkward, I agree. Part of the issue here is that the colors are added to the item's kinda badly, since they're stored in a different setting. And since this component is shared with funnel items and they don't have adjustable colors, it gets really weird. I'll see if there is a nicer way to do this